### PR TITLE
Avoid using the GlobalOpenTelemetry.get() and allow for injection of …

### PR DESCRIPTION
**Pull request by sven-carstens**
…ContextPropagators.

# Description

A problem occures when  using opentelemetry 4.0.0-M1 with spring-boot auto configuration.

The autoconfiguration works as expected and the resulting openTelemetry bean is as expected.
But when the first request comes in, the autoconfigured bean is not used and another openTelemetry configured with **OpenTelemetry.noop()** is used.

**org.apache.camel.opentelemetry.OpenTelemetryTracer#startExchangeBeginSpan** uses **GlobalOpenTelemetry.get()**, which in turn looks into it's **io.opentelemetry.api.GlobalOpenTelemetry#globalOpenTelemetry** which was not set and so it creates the noop implementation.

First step to fix without side effects is to allow the explicit setting of the ContextPropagators which then can be used by **org.apache.camel.opentelemetry.starter.OpenTelemetryAutoConfiguration#openTelemetryEventNotifier** to use the proper autoconfigured instances.

https://camel.zulipchat.com/#narrow/stream/257301-camel-spring-boot/topic/opentelemetry/near/327478493

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
Probably this needs to go on 3.x and 3.20.x too?

**Comment by sven-carstens** [comment]
As we only moved from zipkin/sleuth to opentelemetry while upgrading to Spring-Boot 3 I have no clue how it is handeld with Camel 3.x / Spring-Boot 2.x

**Comment by bvahdat** [comment]
Thanks for your PR.

As you have ticked the following question:

>If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

Would you mind to create a JIRA for this? Then you could reference that JIRA ticket number on this issue as title which would help folks to click on it, see the rational and reasoning behind it when this PR gets merged. Like [this one](https://github.com/apache/camel/commit/824080960f11bccb34907fdd56bcfd788647b618).

**Review by davsclaus — APPROVED** [review]

**Comment by sven-carstens** [comment]
> Would you mind to create a JIRA for this?

I don't have a JIRA account yet.

**Comment by bvahdat** [comment]
> > Would you mind to create a JIRA for this?
> 
> I don't have a JIRA account yet.

Please see here:

https://infra.apache.org/jira-guidelines.html#who
https://infra.apache.org/jira-guidelines.html#account

So all you need is to send an email to `private@camel.apache.org` with the required informations as described :point_up_2:

**Comment by oscerd** [comment]
@sven-carstens I created this https://issues.apache.org/jira/browse/CAMEL-19051

You can reference the issue in the commit message so we could track it.

**Comment by davsclaus** [comment]
We can adjust the commit message on github so I just added the jira number

**Comment by bvahdat** [comment]
@davsclaus I had one review feedback which has not been applied before merging. What are your thoughts on that?

**Comment by davsclaus** [comment]
Did not see any. Also you can just do the change yourself and send a PR if you want feedback.

**Comment by bvahdat** [comment]
> Did not see any.

Is the review feedback :point_up_2: not visible for you?

Screen Shot 2023-02-15 at 15 40 48

Also I don't see myself by the list of the reviewers of this PR on top right. So maybe I am doing something wrong while providing review feedback to improve a given PR?

I am asking because the same thing happened by the following PR:

https://github.com/apache/camel/pull/8934

So why I had to provide a follow-up PR for that:

https://github.com/apache/camel/pull/8911

Which actually had to be done by the original author of [CAMEL-18661](https://issues.apache.org/jira/browse/CAMEL-18661). Don't you think so?

> Also you can just do the change yourself and send a PR if you want feedback.

I think it makes more sense for the original author of a JIRA to apply suggestions, instead of a reviewer to apply his improvement suggestions afterwards as a separate PR as it happened above.

So probably I am doing something wrong while providing feedback to improve a given PR?

**Comment by oscerd** [comment]
I guess the problem is that you review is pending, you need to complete the review you started or add only a comment, in this way you'll be the only one seeing the review.

**Comment by bvahdat** [comment]
Thanks @oscerd but I don't get your point. I have already provided a long comment about what / how to improve.

> you need to complete the review you started 

How should I do that? I though in Github the PR provider would adjust the code according to the provided suggestion and would then mark a given review as "resolved".

**Comment by oscerd** [comment]
The comment you posted as pending, it's not showing up in the PR i'm looking at. So I think you never completed the review you started.

**Comment by oscerd** [comment]
Now I see it. Perfect!

File: components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryTracer.java [hunk 1]
```diff
@@ -86,6 +96,23 @@ protected void initTracer() {
             // No tracer is available, so setup NoopTracer
             tracer = OpenTelemetry.noop().getTracer(instrumentationName);
         }
+
+        if (contextPropagators == null) {
```
[the remark this answers is not in the document]
Comment (sven-carstens) [reply components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryTracer.java:100]: I will change it tomorrow.

[the remark this answers is not in the document]
Comment (sven-carstens) [reply components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryTracer.java:100]: Anyway to re-open thisPR or should I open up a new one?

[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryTracer.java:100]: New PR against main branch

[the remark this answers is not in the document]
Comment (sven-carstens) [reply components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryTracer.java:100]: https://github.com/apache/camel/pull/9358

**Pull request merged**